cmd/admissioncontroller/setup.go

```diff
@@ -0,0 +1,206 @@
+			APIGroups:   []string{"apps"},
+			APIVersions: []string{"v1"},
+			Resources:   []string{"daemonsets"},
+		},
+		{
+			ServicePath: "/replicasets",
+			Name:        "replicasets.aro-admission-controller.redhat.com",
+			Operations:  []admissionregistration.OperationType{"CREATE", "UPDATE"},
+			APIGroups:   []string{"apps"},
+			APIVersions: []string{"v1"},
+			Resources:   []string{"replicasets"},
+		},
+		{
+			ServicePath: "/statefulsets",
+			Name:        "statefulsets.aro-admission-controller.redhat.com",
+			Operations:  []admissionregistration.OperationType{"CREATE", "UPDATE"},
+			APIGroups:   []string{"apps"},
+			APIVersions: []string{"v1"},
+			Resources:   []string{"statefulsets"},
+		},
+		{
+			ServicePath: "/jobs",
+			Name:        "jobs.aro-admission-controller.redhat.com",
+			Operations:  []admissionregistration.OperationType{"CREATE", "UPDATE"},
+			APIGroups:   []string{"batch"},
+			APIVersions: []string{"v1"},
+			Resources:   []string{"jobs"},
+		},
+		{
+			ServicePath: "/cronjobs",
+			Name:        "cronjobs.aro-admission-controller.redhat.com",
+			Operations:  []admissionregistration.OperationType{"CREATE", "UPDATE"},
+			APIGroups:   []string{"batch"},
+			APIVersions: []string{"v1beta1"},
+			Resources:   []string{"cronjobs"},
+		},
+		{
+			ServicePath: "/deploymentconfigs",
+			Name:        "deploymentconfigs.aro-admission-controller.redhat.com",
+			Operations:  []admissionregistration.OperationType{"CREATE", "UPDATE"},
+			APIGroups:   []string{"apps"},
+			APIVersions: []string{"v1"},
+			Resources:   []string{"deploymentconfigs"},
+		},
+		/*		{ //TODO
+				ServicePath: "/deployments",
+				Name:        "deployments.aro-admission-controller",
+				Operations:  []admissionregistration.OperationType{"CREATE", "UPDATE"},
+				APIGroups:   []string{"?"},
+				APIVersions: []string{"?"},
+				Resources:   []string{"deployments"},
+			},*/
+	}
+	failurePolicy := admissionregistration.FailurePolicyType("Fail")
+	vwc := admissionregistration.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "aro-admission-controller.redhat.com",
+		},
+		Webhooks: []admissionregistration.Webhook{},
+	}
+
+	for _, h := range hookconfig {
+		vwc.Webhooks = append(vwc.Webhooks, admissionregistration.Webhook{ClientConfig: admissionregistration.WebhookClientConfig{
+			Service: &admissionregistration.ServiceReference{
+				Name:      "aro-admission-controller",
+				Namespace: "kube-system",
+				Path:      &h.ServicePath,
+			},
+		},
+			FailurePolicy: &failurePolicy,
+			Name:          h.Name,
+			Rules: []admissionregistration.RuleWithOperations{
+				{
+					Operations: h.Operations,
+					Rule: admissionregistration.Rule{
+						APIGroups:   h.APIGroups,
+						APIVersions: h.APIVersions,
+						Resources:   h.Resources,
+					},
+				},
+			},
+		})
+	}
+	return &vwc
+}
+
+func initializeClusterRoleBinding() *authorizationapiv1.ClusterRoleBinding {
```

Comment:
why can't this be done in the sync pod?

Reply:
Dependencies during cluster creation. I could place the logic to check for the aro-admission-controller service and 3 aro-admission-controller pods there.

The role binding shouldn't appear before the ValidatingWebhookConfiguration is in place, which requires the aro-admission-controller Service and Pods to be up.